### PR TITLE
Fixes focus indicator on input checkbox for Firefox

### PR DESCRIPTION
**Pull request by alden-ilao**
## References

#14354 

## Code changes
* updated packages/apputils/style/dialog.css:124
* packages/ui-components/style/styling.css:49

- outline displayed on focus for checkboxes now visible on all platforms
- styling for checkbox:focus now sourced by single file instead of two (packages/ui-components/style/styling.css)

## User-facing changes
The user will now see outlines when hovering checkboxes on all platforms
Before (Firefox): 
image
Before (Chrome):
image
After (Firefox):
image
After (Chrome):
image

### CODE REVIEW

**Comment by krassowski** [comment]
Thank you! Can you confirm that it still works in the dialog; for example you can try clicking on the kernel name to show change kernel dialog, this is how it looks before:

image

Based on the CI results it looks this will be reflected in the snapshots, let's get them updated:

bot please update galata snapshots

**Comment by alden-ilao** [comment]
Thanks for the quick reply! I've been meaning to ask where you are able to get the checkboxes in the dialogs

here's what i got:
image
image

File: packages/apputils/style/dialog.css [hunk 1]
```diff
@@ -121,11 +121,6 @@ button.jp-Dialog-close-button {
   padding-right: 5px;
 }
 
-.jp-Dialog-checkbox > input:focus-visible {
-  outline: 1px solid var(--jp-input-active-border-color);
-  outline-offset: 1px;
-}
-
```
Comment (alden-ilao) [inline packages/apputils/style/dialog.css:128]: (Specifically for checkboxes in dialogs)
for context, this styling is applied by `packages/apputils/src/dialog.tsx` by using the label class as its initial selector.

image

But further down the execution create footer calls `Styling.style(footer.node);` which adds `jp-mod-styled` to the checkbox element. Deleting this styling in`packages/apputils/style/dialog.css` reduces code duplication.
image

**Comment by krassowski** [comment]
bot please update galata snapshots

File: packages/ui-components/style/styling.css [hunk 2]
```diff
@@ -46,6 +46,11 @@ input.jp-mod-styled:focus {
   box-shadow: inset 0 0 4px var(--md-blue-300);
 }
 
+input[type='checkbox'].jp-mod-styled:focus {
```
Comment (krassowski) [inline packages/ui-components/style/styling.css:49]:
```suggestion
input[type='checkbox'].jp-mod-styled:focus-visible {
```

Comment (krassowski) [reply packages/ui-components/style/styling.css:49]: It feels like we should only show the indicator when browser believes that the focus is applicable.

**Review by krassowski — APPROVED** [review]
Thank you @alden-ilao, this looks good to me.

Sorry for back-and-forth with updating the snapshots - when preserving `:focus-visible` approach updates to snapshots were not needed.

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.0.x

**Comment by alden-ilao** [comment]
@krassowski  Apologies for the late reply, I did not have access computer the past couple days, thank you for reviewing and making the changes!

**Pull request merged**